sdk/core/azure-core/azure/core/tracing/ext/opencensus.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+from opencensus.trace import execution_context
+from opencensus.trace import tracer as tracer_module, Span
+from opencensus.trace.propagation import trace_context_http_header_format
+from opencensus.trace.samplers import ProbabilitySampler
+from opencensus.trace.tracers.noop_tracer import NoopTracer
+
+
+class OpencensusWrapper:
+    def __init__(self, span=None, name="parent_span"):
+        # type: (Any) -> None
+        tracer = self.get_current_tracer()
+        self.was_created_by_azure_sdk = False
+        if span is None:
+            instrumentation_key = self._get_environ("APPINSIGHTS_INSTRUMENTATIONKEY")
+            prob = self._get_environ("AZURE_TRACING_SAMPLER") or 0.001
+            if tracer is None or isinstance(tracer, NoopTracer):
+                if instrumentation_key is not None:
+                    from opencensus.ext.azure.trace_exporter import AzureExporter
+
+                    tracer = tracer_module.Tracer(
+                        exporter=AzureExporter(instrumentation_key=instrumentation_key),
+                        sampler=ProbabilitySampler(prob),
+                    )
+                else:
+                    tracer = tracer_module.Tracer(sampler=ProbabilitySampler(prob))
+                self.was_created_by_azure_sdk = True
+            span = tracer.span(name=name)
+
+        self.tracer = tracer
+        self.trace_id = None
+        if span.context_tracer:
+            self.trace_id = span.context_tracer.span_context.trace_id
+        self.span_instance = span
```

Comment:
given how for you have to dig down to obtain `trace_id` I'd suggest considering not duplicating it outright, instead consider having a `trace_id` property method that returns `self.span_instance.context_tracer.span_context.trace_id`

Reply:
Also `span_instance` could probably be private `self._span_instance`

Reply:
I just got rid of trace and span id as I never really used it. span_instance is needed when I set the context for opencensus later on. 

https://github.com/SuyogSoti/azure-sdk-for-python/blob/7f35de4b250f1019ddf607dd4da5e9b7e1fc17b8/sdk/core/azure-core/azure/core/trace/base.py#L11

and

https://github.com/SuyogSoti/azure-sdk-for-python/blob/7f35de4b250f1019ddf607dd4da5e9b7e1fc17b8/sdk/core/azure-core/azure/core/trace/base.py#L66-L80

Reply:
resolved in this discussion: https://github.com/Azure/azure-sdk-for-python/pull/6264#discussion_r302170869